pennylane/qnodes/base.py

```diff
@@ -515,14 +518,15 @@ def _construct(self, args, kwargs):
             # generate the program queue by executing the quantum circuit function
             if self.mutable:
                 # it's ok to directly pass auxiliary arguments since the circuit is re-constructed each time
-                # (positional args must be replaced because parameter-shift differentiation requires Variables)
+                # (positional args must be replaced because parameter-shift differentiation requires VariableRefs)
                 res = self.func(*self.arg_vars, **kwargs)
             else:
                 # TODO: Maybe we should only convert the kwarg_vars that were actually given
                 res = self.func(*self.arg_vars, **self.kwarg_vars)
         except:
             # If there was an error in the function call it could be that the parameters were corrupted
-            # In this case we wipe our Variable cache
+            # In this case we wipe our VariableRef cache
+            # FIXME if this can happen that's a bug that needs to be fixed
```

Comment:
I disagree. This happens for example when somebody uses the following code:
```python
@qml.qnode(dev)
def circuit(arr):
    qml.RX(arr[0], wires=[0])
    qml.RX(arr[1], wires=[1])
    return qml.expval(qml.PauliZ(0))

try:
    circuit([1.344]) # Error because arr[1] does not exist
except:
    print("failed")

circuit([1.344, 3.444]) # Works, but would fail if we didn't wipe the cach
```

Reply:
Ahh, okay, I see it now. An incomplete construction should leave nothing behind.